test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -472,6 +472,32 @@ def test_insufficient_sharding_dims(self):
             self.assertEqual([1, 20], shard_metadata.shard_lengths)
             self.assertEqual(f'rank:{shard_rank}/cuda:{shard_rank}', shard_metadata.placement)
 
+    @with_comms
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_tensor_sizes(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+
+        # Test with *args
```

Comment:
[Optional] Nit: you can loop over `_sharded_tensor.empty(spec, ...)` to reduce the redundant code `self.assertEqual(torch.Size([10, 20]), sharded_tensor.size())`. It probably won't save too much though, so totally up to you.

Reply:
Not sure how this would work concisely, the loop somehow would need to iterate of things like 10, 20, [10, 20] and (10, 20) separately.